proof/crefine/RISCV64/Syscall_C.thy

```diff
@@ -899,41 +921,80 @@ sorry (* FIXME RT: decodeInvocation_ccorres
     apply csymbr
     apply (rule ccorres_trim_returnE, simp+)
     apply (rule ccorres_call,
-           erule decodeIRQHandlerInvocation_ccorres, simp+)
-   apply (rule ccorres_inst[where P=\<top> and P'=UNIV])
+           erule decodeIRQHandlerInvocation_ccorres, simp+)[1]
+   apply (rule ccorres_Cond_rhs)
+    apply (simp add: if_to_top_of_bind)
+    apply (rule ccorres_trim_returnE, simp+)
+    apply (simp add: if_to_top_of_bind Collect_const[symmetric]
+                del: Collect_const)
+    apply (rule ccorres_if_cond_throws2[where Q=\<top> and Q'=\<top>])
+       apply (clarsimp simp: isCap_simps)
+      apply (simp add: throwError_bind invocationCatch_def)
+      apply (rule syscall_error_throwError_ccorres_n)
+      apply (simp add: syscall_error_to_H_cases)
+     apply (simp only: liftME_invocationCatch comp_def id_def)
+     apply (rule ccorres_call, rule decodeSchedControlInvocation_ccorres[where cp=cp])
+        apply assumption
+       apply (simp+)[3]
+    apply vcg
+   apply (rule ccorres_Cond_rhs)
+    apply (simp add: if_to_top_of_bind)
```

Comment:
seeing a lot of `if_to_top_of_bind` ... are there multiple ifs in here, or is something dodgy with automation?